### PR TITLE
7.0 travis coverage

### PR DESCRIPTION
**Pull request by yvaucher**

### CODE REVIEW

**Comment by eLBati** [comment]
I'm not able to reproduce the errore showed here
https://travis-ci.org/OCA/account-invoice-reporting/builds/29119964#L1736

Running tests of account_invoice_production_lot does not raise errors

Moreover, trying to run

```
$ bin/start_odoo-7 --test-enable --log-level=test -i account_draft_invoice_print,account_invoice_delivery_address,account_invoice_production_lot,html_invoice_product_note,invoice_print_report_balance_payment,invoice_report_assemble,invoice_webkit,base_report_assembler,report_xls,picking_dispatch,picking_dispatch_wave,product_serial,stock_cancel,stock_filter_none_zero_qty,stock_move_backdating,stock_picking_invoice_link,stock_picking_priority,stock_picking_show_returns,stock_split_picking,base_headers_webkit,report_webkit_chapter_server -d demo_7
```

with buildout with

```
version = git http://github.com/odoo/odoo.git odoo 7.0
addons = git https://github.com/OCA/reporting-engine.git parts/reporting-engine 7.0
         git https://github.com/OCA/stock-logistics-workflow.git parts/stock-logistics-workflow 7.0
         git https://github.com/OCA/webkit-tools.git parts/webkit-tools 7.0
         git https://github.com/yvaucher/account-invoice-reporting.git parts/yvaucher-account-invoice-reporting 7.0-travis-coverage
```

I get

```
2014-07-04 11:30:12,876 17637 INFO demo_7 openerp.modules.loading: module picking_dispatch_wave: loading test/test_dispatch_wave.yml
[...]
2014-07-04 11:30:14,353 17637 TEST demo_7 openerp.tools.yaml_import: Then I confirm the demand
[...]
  File "/home/elbati/workspace/odoo/buildout_cfg-odoo-7/parts/stock-logistics-workflow/picking_dispatch_wave/dispatch_wave.py", line 55, in _get_moves_from_picking_list
    context=context)
  File "/home/elbati/workspace/odoo/buildout_cfg-odoo-7/parts/odoo/openerp/osv/orm.py", line 2372, in search
    return self._search(cr, user, args, offset=offset, limit=limit, order=order, context=context, count=count)
  File "/home/elbati/workspace/odoo/buildout_cfg-odoo-7/parts/odoo/openerp/osv/orm.py", line 4937, in _search
    query = self._where_calc(cr, user, args, context=context)
  File "/home/elbati/workspace/odoo/buildout_cfg-odoo-7/parts/odoo/openerp/osv/orm.py", line 4765, in _where_calc
    where_clause, where_params = e.to_sql()
  File "/home/elbati/workspace/odoo/buildout_cfg-odoo-7/parts/odoo/openerp/osv/expression.py", line 1220, in to_sql
    q, p = self.__leaf_to_sql(leaf)
  File "/home/elbati/workspace/odoo/buildout_cfg-odoo-7/parts/odoo/openerp/osv/expression.py", line 1145, in __leaf_to_sql
    raise ValueError("Invalid domain term %r" % (leaf,))
ValueError: Invalid domain term ('picking_id', 'in', set([34]))
```

still no errors about account_invoice_production_lot

Got the same error running

```
bin/start_odoo-7 --test-enable --log-level=test -i account_draft_invoice_print,account_invoice_delivery_address,html_invoice_product_note,invoice_print_report_balance_payment,invoice_report_assemble,invoice_webkit,base_report_assembler,report_xls,picking_dispatch,picking_dispatch_wave,product_serial,stock_cancel,stock_filter_none_zero_qty,stock_move_backdating,stock_picking_invoice_link,stock_picking_priority,stock_picking_show_returns,stock_split_picking,base_headers_webkit,report_webkit_chapter_server,account_invoice_production_lot -d demo_7
```

(installing account_invoice_production_lot last)

I did not investigate more because I suspect it depends on having several addons (potentially conflicting) tested together.
Indeed I agree with what Guewen said about testing many modules together
http://openerp-community.2306076.n4.nabble.com/Openerp-community-OCA-transition-from-Launchpad-to-Github-tp4645893p4646141.html
I think we should test the modules individually.

**Comment by yvaucher** [comment]
@eLBati Ok thanks a lot for your time tested this. Ok it seems we have no choice but to separate tests in this PR. @nbessi is already working on this

**Comment by eLBati** [comment]
I was looking at
https://github.com/OCA/maintainer-quality-tools/blob/master/travis/travis_run_tests
and I think a solution could be to use the 'DB_NAME' parameter of 'travis_run_tests' as the module name too.
So, the .travis.yml file could just run the single test on different DBs calling travis_run_tests for every module.
Something like

```
travis_run_tests 7.0 account_invoice_production_lot $HOME/reporting-engine $HOME/stock-logistics-workflow $HOME/webkit-tools
travis_run_tests 7.0 picking_dispatch_wave $HOME/reporting-engine $HOME/stock-logistics-workflow $HOME/webkit-tools
```

Here
https://github.com/OCA/maintainer-quality-tools/pull/15
is what I was thinking about.

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+python:
+  - "2.7"
+
+virtualenv:
+  system_site_packages: true
+
+install:
+ - git clone https://github.com/OCA/maintainer-quality-tools.git $HOME/maintainer-quality-tools
+ - export PATH=$HOME/maintainer-quality-tools/travis:$PATH
+ - $HOME/maintainer-quality-tools/travis/travis_install_nightly 7.0
```
Comment (bwrsandman) [inline .travis.yml]: This could be simply `travis_install_nightly 7.0`

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+ - pip install coveralls flake8
+ - pip install PyPDF2==1.18
+ - git clone https://github.com/OCA/reporting-engine -b 7.0 $HOME/reporting-engine
+ - git clone https://github.com/OCA/stock-logistics-workflow -b 7.0 $HOME/stock-logistics-workflow
+ - git clone https://github.com/OCA/webkit-tools -b 7.0 $HOME/webkit-tools
+
+services:
+  - postgresql
```
Comment (bwrsandman) [inline .travis.yml]: `services` section is not necessary

**Comment by yvaucher** [comment]
Thanks for the review @bwrsandman 

made the changes

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+ - travis_install_nightly 7.0
+ - pip install coveralls flake8
+ - pip install PyPDF2==1.18
+ - git clone https://github.com/OCA/reporting-engine -b 7.0 $HOME/reporting-engine
+ - git clone https://github.com/OCA/stock-logistics-workflow -b 7.0 $HOME/stock-logistics-workflow
+ - git clone https://github.com/OCA/webkit-tools -b 7.0 $HOME/webkit-tools
+
+script:
+    - travis_run_flake8
+    - travis_run_tests 7.0 openerp_test $HOME/reporting-engine $HOME/stock-logistics-workflow $HOME/webkit-tools
```
Comment (bwrsandman) [inline .travis.yml]: You can take out `openerp_test` is it no longer necessary

**Comment by yvaucher** [comment]
@bwrsandman done

**Comment by bwrsandman** [comment]
:+1: for coverage, travis and README.md

Is there a plan to fix the failing unittest?

**Comment by yvaucher** [comment]
I'm testing with encapsulated tests: https://github.com/OCA/account-invoice-reporting/pull/6 according to @eLBati tests are rights. This is just a conflict with multiple modules

**Comment by yvaucher** [comment]
Thanks to @eLBati This is green !

**Pull request merged**